src/components/Tabs/ScatterPlotTab.tsx

```diff
@@ -147,6 +151,8 @@ export default memo(function ScatterPlotTab(props: ScatterPlotTabProps): ReactEl
     }
   }, [props.scatterPlotConfig]);
   const { xAxis: xAxisFeatureKey, yAxis: yAxisFeatureKey, rangeType } = scatterConfig;
+  const [xAxisData, setXAxisData] = useState<RawFeatureData | null>(null);
+  const [yAxisData, setYAxisData] = useState<RawFeatureData | null>(null);
```

Comment:
Similar pattern to `Viewer.tsx`, where we save the loaded feature data as state instead of only storing the key.